artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/CoreAmqpConverter.java

```diff
@@ -355,6 +359,15 @@ public static AMQPMessage fromCore(ICoreMessage coreMessage) throws Exception {
       }
    }
 
+   private static String removePrefix(String address) {
```

Comment:
Isn’t this logic somewhere in a util or static already? Looks familiar

Reply:
I looked for it and couldn't find it. I can check tomorrow what the client is doing here.

Reply:
PrefixUtil

Reply:
ServerSessionImpl.removePrefix invokes it, you could probably invoke it via the same server session, that way it would pick up the same config as server session already has done for you.

Reply:
I had to create a new method though.